### ISSUE TITLE
very sensitive to params.libsonnet formatting

### ISSUE THREAD

**Issue by redbaron** [ksonnet/ksonnet#290]
/kind bug

**What happened**:

certain format of `params.libsonnet` upsets ksonnet

**How to reproduce it (as minimally and precisely as possible)**:

with following components/params.libsonnet:
```
{components: { }}
```
ksonnet complains:
```
# Writing component at 'components/generic-app'
# Error reading /tmp/tmp.pF34rMG0Lb/app/environments/default/main.jsonnet: /tmp/tmp.pF34rMG0Lb/app/components/params.libsonnet:1:18-19 Did not expect: ":"
#
#     "generic-app": {
```

Changing params file to (adding newline):
```
{components: {
}}
```

makes it happy again

**Environment**:
- ksonnet version (use `ks version`):  v0.8.0

**Comment by bryanl** [comment]
This scenario is caused by the way the params file is parsed. It isn't updated by string manipulate rather than running through the lexer/parser. When params moves to the lexer/parser path for all operations, it should work with any valid file.

**Comment by bryanl** [comment]
This problem should be resolved in 0.10+. I'll leave it open for a little while for validation.
